src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -928,6 +927,15 @@ private BoundCollectionExpression ConvertCollectionExpression(
             }
             else
             {
+                // PROTOTYPE: Add support for interfaces and Create methods.  Keep diagnostic for arrays/spans.
```

Comment:
interfaces coming in https://github.com/dotnet/roslyn/pull/80554.  'Ccreate' methods will come in a PR after that.